src/tif_model_iterator/tif_kernel_iterator.py

```diff
@@ -267,54 +335,107 @@ def transform_to_polygons(df: pd.DataFrame) -> gpd.GeoDataFrame:
+                    rd_x_ul,
+                    rd_y_ul,
+                    rd_x_ur,
+                    rd_y_ur,
+                    rd_x_lr,
+                    rd_y_lr,
+                    rd_x_ll,
+                    rd_y_ll,
+                )
+            ]
+
+            df = df[["geometry", "label"]]
+
+            gdf = gpd.GeoDataFrame(df, geometry=df.geometry)
+            gdf = gdf.set_crs(epsg=self.input_crs)
+
+            if self.output_crs:
+                gdf = gdf.to_crs(self.output_crs)
+
+            print("Geometry made in: " + str(timer() - start) + " second(s)")
+        else:
+            df = df[["rd_x", "rd_y", "label"]]
+            gdf = pd.DataFrame(df)
+            print("Pandas dataframe made in: " + str(timer() - start) + " second(s)")
 
-        gdf = gpd.GeoDataFrame(df, geometry=df.geometry)
-        gdf = gdf.set_crs(epsg=28992)
-        print("Geometry made in: " + str(timer() - start) + " second(s)")
         return gdf
 
-    def _write_part_to_file(
-        self,
-        gdf: gpd.GeoDataFrame,
-        step: int,
-    ):
+    def _write_part_to_file(self, gdf, step: int):
         """
         Writes a part of the tif file to an outputfile.
 
         @param gdf: GeoDataFrame to write to part file
         @param step: the step currently being written to file
         """
-        print("Dissolving and writing to file")
 
-        output_file_name = self.output_file_name_generator.generate_part_output_path(
-            step
-        )
+        if self.dissolve_parts:
+            print("Dissolving and writing part to file")
 
-        dissolve_gpd_output(gdf, output_file_name)
+            output_file_name = (
+                self.output_file_name_generator.generate_part_output_path(step)
+            )
+
+            dissolve_gpd_output(gdf, output_file_name)
+        else:
+            print("Writing part to file")
+
+            output_file_name = (
+                self.output_file_name_generator.generate_part_output_path(step)
+            )
+
+            if ".geojson" in output_file_name:
+                gdf.to_file(output_file_name)
+            elif ".csv" in output_file_name:
+                gdf.to_csv(output_file_name, index=False)
+            elif ".parquet" in output_file_name:
+                gdf.to_parquet(output_file_name)
 
     def _write_full_gdf_to_file(self):
         """
         Reads all files of the parts, combines them into 1 and writes the full gdf to file.
         """
+        print("Merging all parts into a final file")
         all_part_files = glob.glob(
             self.output_file_name_generator.glob_wild_card_for_part_extension_only()
         )
-        full_gdf = pd.concat([gpd.read_file(file) for file in all_part_files])
+
+        print("Final path will be: " + str(self.final_output_path))
+
+        if ".parquet" in self.final_output_path:
+            full_gdf = pd.concat([pd.read_parquet(file) for file in all_part_files])
```

Comment:
What happened to the geopandas call here? It's not a GeoDataFrame, but a Pandas DataFrame, right? Isn't that problematic further on?

Reply:
Okay clear!